scipy/stats/tests/test_generation/reference_distributions.py

```diff
@@ -0,0 +1,286 @@
+    implementations are easily compared against their mathematical definitions.
+    No attempt is made to handle edge cases or be fast, and arbitrary precision
+    arithmetic is trusted for accuracy rather than making the method
+    implementations "smart".
+
+    Notes
+    -----
+
+    In this infrastructure, distributions families are classes, and
+    fully-specified distributions (i.e. with definte values of all family
+    parameters) are instances of these classes. Typically, the public methods
+    accept as input only the argument at which the at which the function is to
+    be evaluated. Unlike SciPy distributions, they never accept values of
+    distribution family shape, location, or scale parameters. A few
+    other parameters are noteworthy:
+
+    - All methods accept `dtype` to control the output data type. The default
+      is `np.float64`, but `object` or `mp.mpf`
+    - `ppf`/`isf` accept a `guess` because they use a scalar rootfinder
+      to invert the `cdf`/`sf`. This is passed directly into the `x0` method
+      of `mpmath.findroot`; see its documentation for details.
+    - moment accepts `order`, an integer that specifies the order of the (raw)
+      moment, and `center`, which is the value about which the moment is
+      taken. The default is to calculate the mean and use it to calculate
+      central moments; passing `0` results in a noncentral moment. For
+      efficiency, the mean can be passed explicitly if it is already known.
+
+    Follow the example of SkewNormal to generate new reference distributions,
+    overriding only `__init__` and `_pdf`*. Use the reference distributions to
+    generate reference values for unit tests of SciPy distribution method
+    precision and robustness (e.g. for extreme arguments). If the a SciPy
+    methods implementation is independent and yet the output matches reference
+    values generated with this infrastructure, it is unlikely that the SciPy
+    and reference values are both inaccurate.
+
+    * If the SciPy output *doesn't* match and the cause appears to be
+    inaccuracy of the reference values (e.g. due to numerical issues that
+    mpmath's arbitrary precision arithmetic doesn't handle), then it may be
+    appropriate to override a method of the reference distribution rather than
+    relying on the generic implementation. Otherwise, hesistate to override
+    methods: the generic implementations are mathematically correct and easy
+    to verify, whereas an override introduces many possibilities of mistakes,
+    requires more time to write, and requires more time to review.
+
+    In general, do not create custom unit tests to ensure that
+    SciPy distribution methods are *correct* (in the sense of being consistent
+    with the rest of the distribution methods); generic tests take care of
+    that.
+
+
+    """
+
+    def __init__(self, **kwargs):
+        assert mpmath.dps is None
```

Comment:
I might want to do more here, like check that `mpmath.mp.dps` exceeds the default?

Reply:
I missed that there was an `assert` here. I would say fine here because we are only going to run this on our side, but in general it's a bad practice to use `assert` for control flow.

Reply:
Done.